cmd/kubeadm/app/util/etcd/etcd.go

```diff
@@ -118,25 +119,38 @@ func NewFromCluster(client clientset.Interface, certificatesDir string) (*Client
 
 // Sync synchronizes client's endpoints with the known endpoints from the etcd membership.
 func (c *Client) Sync() error {
-	cli, err := clientv3.New(clientv3.Config{
-		Endpoints:   c.Endpoints,
-		DialTimeout: 20 * time.Second,
-		TLS:         c.TLS,
-	})
-	if err != nil {
-		return err
+	var cli *clientv3.Client
+	var err error
+
+	for _, endpoint := range c.Endpoints {
+		cli, err = clientv3.New(clientv3.Config{
+			Endpoints:   []string{endpoint},
+			DialTimeout: 20 * time.Second,
```

Comment:
i think 5 seconds might be sufficient here.

Reply:
I agree, but that was not necessarily in scope of this fix so I left the timeout as before.

Reply:
Leave 20, just because we know the constants work and every time we change them something weird happens.